src/dbnode/persist/fs/commitlog/files.go

```diff
@@ -72,39 +134,46 @@ func ReadLogInfo(filePath string, opts Options) (time.Time, time.Duration, int64
 	err = fd.Close()
 	fd = nil
 	if err != nil {
-		return time.Time{}, 0, 0, err
+		return time.Time{}, 0, 0, fsError{err}
 	}
 
 	return time.Unix(0, logInfo.Start), time.Duration(logInfo.Duration), logInfo.Index, decoderErr
 }
 
 // Files returns a slice of all available commit log files on disk along with
 // their associated metadata.
-func Files(opts Options) ([]File, error) {
+func Files(opts Options) ([]FileOrError, error) {
 	commitLogsDir := fs.CommitLogsDirPath(
 		opts.FilesystemOptions().FilePathPrefix())
 	filePaths, err := fs.SortedCommitLogFiles(commitLogsDir)
 	if err != nil {
 		return nil, err
 	}
 
-	commitLogFiles := make([]File, 0, len(filePaths))
+	commitLogFiles := make([]FileOrError, 0, len(filePaths))
 	for _, filePath := range filePaths {
+		file := File{
+			FilePath: filePath,
+		}
+
 		start, duration, index, err := ReadLogInfo(filePath, opts)
-		if err != nil {
+		if _, ok := err.(fsError); ok {
 			return nil, err
 		}
 
-		commitLogFiles = append(commitLogFiles, File{
-			FilePath: filePath,
-			Start:    start,
-			Duration: duration,
-			Index:    index,
-		})
+		if err == nil {
+			file.Start = start
+			file.Duration = duration
+			file.Index = index
+		}
+
+		commitLogFiles = append(commitLogFiles, NewFileOrError(
+			file, err, filePath))
 	}
 
 	sort.Slice(commitLogFiles, func(i, j int) bool {
-		return commitLogFiles[i].Start.Before(commitLogFiles[j].Start)
+		// Sorting is best effort here since we may not know the start.
```

Comment:
why sort at all? and if you're going to sort - you can sort errors first (or last) and then guarantee order too. 

or better yet change the return type for this function: `([]File, []FileWithError, error)` and then won't need the FileOrError type.

Reply:
Sorting is important because you want to try and read the commit log files in order in the bootstrapper

Reply:
Your interface change suggestion is good though, just made it